app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -277,8 +279,8 @@ class BrowserTabViewModel(
         object HideKeyboard : Command()
         class ShowFullScreen(val view: View) : Command()
         class DownloadImage(val url: String, val requestUserConfirmation: Boolean) : Command()
-        class ShowSavedSiteAddedConfirmation(val savedSite: SavedSite) : Command()
-        class ShowEditSavedSiteDialog(val savedSite: SavedSite) : Command()
+        class ShowSavedSiteAddedConfirmation(val savedSite: SavedSite, val bookmarkFolder: BookmarkFolder?) : Command()
```

Comment:
Not sure about this approach: we are using a sealed class (`savedSite`) to represent a Bookmark/Favorite, but now we have a second param that only applies to Bookmarks (`bookmarkFolder`). If that's the case, it should be part of the sealed class as well, because it only applies to one of the types.

`SavedSite` is a domain (layer) sealed class.

In here, I think you are passing `bookmarkFolder` to show bookmark folder name in the dialog. If that's the case, my approach will be to create a sealed class specific for those dialog. Then you can map your savedSite into your new saved site and use it there. But please, double check this approach, I could be missing something.

Reply:
You are right.
I've now used a separate data class for the change in saved site to process interaction with the views.